CppCoreGuidelines.md

```diff
@@ -6436,8 +6436,13 @@ A non-throwing move will be used more efficiently by standard-library and langua
     template<typename T>
     class Vector {
     public:
-        Vector(Vector&& a) noexcept :elem{a.elem}, sz{a.sz} { a.sz = 0; a.elem = nullptr; }
-        Vector& operator=(Vector&& a) noexcept { elem = a.elem; sz = a.sz; a.sz = 0; a.elem = nullptr; }
+        Vector(Vector&& a) noexcept :elem{a.elem}, sz{a.sz} { a.elem = nullptr; a.sz = 0; }
+        Vector& operator=(Vector&& a) noexcept {
+            delete elem;
+            elem = a.elem; a.elem = nullptr;
```

Comment:
seems like a fit for `std::exchange`?

Reply:
I thought so too; you could use it in a number of places in this document, but I it would increase the complexity of examples because not everyone knows `std::exchange`.

Reply:
I think that may be a topic for another PR. This one is not about improving code style, just fixing a broken example.

Reply:
fair enough